ferretdb/ferretdb.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ferretdb
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn"
+	"github.com/FerretDB/FerretDB/internal/handlers/registry"
+	"github.com/FerretDB/FerretDB/internal/util/logging"
+)
+
+// Config contains a backend connection strings.
+type Config struct {
+	Handler     string
+	PostgresURL string
+	TigrisURL   string
+}
+
+// FerretDB proxy.
+type FerretDB struct {
+	config Config
+}
+
+// New registers backend handler and returns a new FerretDB.
+func New(conf Config) FerretDB {
+	switch conf.Handler {
+	case "pg":
+		registry.RegisterPg(conf.PostgresURL)
+	case "tigris":
+		registry.RegisterTigris(conf.TigrisURL)
+	default:
+		panic(fmt.Sprintf("Unknown backend handler %q.", conf.Handler))
+	}
+	return FerretDB{
+		config: conf,
+	}
+}
+
+// GetConnectionString returns the backend connection string.
+func (fdb *FerretDB) GetConnectionString() string {
+	return "mongodb://127.0.0.1:27017"
+}
+
+// Run runs the FerretDB proxy as a library with:
+// * error level logging
+// * monitoring disabled
+// * handler PostgreSQL.
+func (fdb *FerretDB) Run(ctx context.Context) error {
+	listenAddr := "127.0.0.1:27017"
+	mode := clientconn.NormalMode
+	testConnTimeout := time.Duration(0)
+
+	logging.Setup(zapcore.ErrorLevel)
+	logger := zap.L()
```

Comment:
The user of our library might prefer something else but `zap`. Let's add `logger` to the config as the minimal needed interface, not the exact choice?

Reply:
> The user of our library might prefer something else but `zap` or might want to use `zap` but with some specific configuration. Let's add `logger` to the config as the minimal needed interface, not the exact choice? (and so, our library won't depend on `zap` at all)

I would suggest to add logger in another PR.

Reply:
Then please create an issue or add this item to DoD of the current issue.

One other possible solution - currently we might not need the logger at all. If we just return an error, and there are not many things we can log, let's simplify the code and get rid of zap completely.

Reply:
On the other hand, we transfer loger to a handler... This is huge, yes. That probably needs a separate issue.

Reply:
> let's simplify the code and get rid of zap completely.

If you think of removing zap as a FerretDB dependency completely - It sounds as a big task.

Reply:
> > let's simplify the code and get rid of zap completely.
> 
> If you think of removing zap as a FerretDB dependency completely - It sounds as a big task.

Let's not implement it at this point.

Reply:
This thread needs more clarity.

@AlekSi so, what do we agree about by "not implement"? We don't use it and don't make it a dependency? Or do we keep it as is (a non-configurable dependency)?